File: app/src/main/java/org/kindone/willingtodo/persistence/TaskContextPersistenceProvider.java
package org.kindone.willingtodo.persistence;

import org.kindone.willingtodo.data.TaskContext;

import java.util.List;

/**
 * Created by kindone on 2016. 12. 22..
 */

public interface TaskContextPersistenceProvider {

    List<TaskContext> getTaskContexts();

    int getModeOfTaskContext(long contextId);

    void setModeOfTaskContext(long contextId, int mode);

    TaskContext createTaskContext(TaskContext context);

    void updateTaskContext(TaskContext context);

    void swapPositionOfTaskContexts(long id1, long id2);

    void deleteTaskContext(long id);

    int getVersion();
}

File: app/src/main/java/org/kindone/willingtodo/persistence/TaskContextPersistenceProvider.kt
package org.kindone.willingtodo.persistence

import org.kindone.willingtodo.data.TaskContext

/**
 * Created by kindone on 2016. 12. 22..
 */

interface TaskContextPersistenceProvider {

    val taskContexts: List<TaskContext>

    fun getModeOfTaskContext(contextId: Long): Int

    fun setModeOfTaskContext(contextId: Long, mode: Int)

    fun createTaskContext(context: TaskContext): TaskContext

    fun updateTaskContext(context: TaskContext)

    fun swapPositionOfTaskContexts(id1: Long, id2: Long)

    fun deleteTaskContext(id: Long)

    val version: Int
}
